### PR TITLE
Wraps maliput::api::RoadNetwork to avoid conflict during python bindings

### PR DESCRIPTION
**Pull request by francocipollone**
Resolves https://github.com/maliput/delphyne/issues/846

### CODE REVIEW

**Comment by francocipollone** [comment]
CC: Ready to re-review

File: include/delphyne/roads/road_network.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace roads {
+
+/// Wrapper around a maliput::api::RoadNetwork.
+/// This implementation is needed for the python bindings due to a lack of support when handling smart pointers.
+/// For more information see:
+/// - https://github.com/maliput/delphyne/issues/846
+/// - https://github.com/maliput/maliput_infrastructure/issues/225
+///
+/// TODO(https://github.com/maliput/delphyne/issues/847): Remove this wrapper once the above issues are resolved.
+class RoadNetwork {
```
Comment (agalbachicar) [inline include/delphyne/roads/road_network.h:17]: What if we call it RoadNetworkWrapper? Wouldn't it be more accurate?

Also, by looking at the numerous changes in code with road_network->get()->something I suggest we create the following methods in this class (see https://en.cppreference.com/w/cpp/memory/unique_ptr/operator* ):

```C++
maliput::api::RoadNetwork* operator->() const noexcept {
  return rn_.get();
}
```

Comment (agalbachicar) [reply include/delphyne/roads/road_network.h:17]: We should also add tests to this class.

**Review by agalbachicar — CHANGES_REQUESTED** [review]
PTAL

File: include/delphyne/roads/road_network.h (hunk 1 above)
Comment (francocipollone) [reply include/delphyne/roads/road_network.h]: Done.

File: include/delphyne/roads/road_network_wrapper.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#pragma once
```
Comment (agalbachicar) [inline include/delphyne/roads/road_network_wrapper.h:1]: Missing license note.

**Review by agalbachicar — APPROVED** [review]
LGTM

File: include/delphyne/roads/road_network_wrapper.h (hunk 2 above)
Comment (francocipollone) [reply include/delphyne/roads/road_network_wrapper.h:31]: Done

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**